js/README.md

```diff
@@ -100,55 +100,163 @@ Subdirectories with additional utilities
+<!-- `uniquenameBlock.palette = palettes.dict['yourpalettename'];` -->
+
+<!-- e.g., `pitchNumberBlock.palette = palettes.dict['pitch'];` -->
+e.g. `this.setPalette('yourPalleteName);`
+
+* Your class definition should look similar to this:
 
-e.g., `pitchNumberBlock.palette = palettes.dict['pitch'];`
+```
+  class NamedDoBlock extends FlowBlock{
+    constructor() {
+      super(nameddo);
+      this.setPalette('action');
+    }
+  }
+```
+Note: After the new update there is no requirement for a `beginnerMode` check as `BaseBlock` automatically performs that check. 
 
 The palette can be any of the palettes listed in `turtledef.js`. The
 color of the block is defined by the palette used.
 
-3. To add block to the protoblock dictionary
+3. Add a call to `new myNewBlock.setup()` in the previously defined `setup` function.
+
+e.g. `function setupActionBlocks() {`
+
+  `new CalcBlock().setup();`
+
+`}`
+
+* For arg blocks, define a function `arg` inside the block class definition. There are 4 arguments currently passed to this function viz. `(logo, turtle, blk, receivedArg)`.
+
+e.g. 
+
+```
+  class CalcBlock extends LeftBlock {
+    constructor() {
+        super('calc');
+        this.setPalette('action');
+    }
+
+    arg(logo, turtle, blk, receivedArg) {
+    }
+}
+```
+
+* For flow bocks define a function `flow` on the block. The same 4 arguments are passed to the flow function currently: `(logo, turtle, blk, receivedArg)`.
+
+e.g 
+```
+class ListenBlock extends FlowBlock {
+    constructor() {
+        super('listen');
+        this.setPalette('action');
+    }
+
+    flow(args, logo, turtle, blk, receivedArg) {
+    }
+}
+```
+
+Note: Trailing arguments can be neglected in both functions, if not needed.
+
+<!-- To add block to the protoblock dictionary
 
 `blocks.protoBlockDict['uniquename'] = uniquenameBlock;`
+ -->
+<!-- e.g., `blocks.protoBlockDict['pitchnumber'] = pitchNumberBlock;` -->
+
+4. Write the logic for the block in either of the two functions, `arg()` or `flow()`.
+
+Note: Change of syntax after MVC structure:
+
+* `that.` used while writing logic in `logo.js` is to be replaced with `logo.`
+* For arg blocks calls to `that.blocks.blockList[blk].value = ` are to be replace with a `return` statement.
+* In case of flow blocks, return value should be in the form `[childFlow, childFlowCount]` or `[]` if they are unchanged. 
 
-e.g., `blocks.protoBlockDict['pitchnumber'] = pitchNumberBlock;`
+So changes to these variabled should be checked and return keyword should be used.
 
-4. Define additional block properties, e.g.,
+e.g. 
+```
+class StartBlock extends StackClampBlock {
+    constructor() {
+        super('start');
+        this.setPalette('action');
+        
+        this.formBlock({ name: _('start'), canCollapse: true });
+    }
+
+    flow(args) {
+        if (args.length === 1)
+            return [args[0], 1];
+    }
+}
+```
+
+5. In case of flow blocks, blockID should be added to the case stack.
```

Comment:
Not sure what you are talking about here.

Reply:
Does the user require to do anything with the blockId?

Reply:
I still don't understand. What "case stack"? What "blockID"?

Reply:
I might have got confused there. I think I saw something similar in the PR for Block API but it was regarding porting of blocks from `logo.js`. It's not needed here, I'll remove it.